examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,255 @@
+# Nvflare with TensorFlow
+
+This example uses Nvflare to train an image classifier using federated averaging and TensorFlow as the deep learning framework.
+
+[NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) (**NV**IDIA **F**ederated **L**earning **A**pplication **R**untime **E**nvironment)
+is a domain-agnostic, open-source, extensible SDK that allows researchers and data scientists to adapt existing ML/DL workflows to a federated paradigm.
+It enables platform developers to build a secure, privacy-preserving offering for a distributed multi-party collaboration.
+
+[Here](https://nvflare.readthedocs.io/en/main/flare_overview.html#high-level-system-architecture) is a high level system architecture of Nvflare.
+
+> **_NOTE:_** This example uses the [MNIST](http://yann.lecun.com/exdb/mnist/) handwritten digits dataset and will load its data within the trainer code.
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+## High-level diagram
+
+The following diagram shows one server and two clients that are connected to the server through a secure gRPC link:
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the Nvflare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## Deploy the blueprint
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+To deploy the Nvflare demo described in this document, you need to deploy the [Federated learning blueprint](../../../../README.md#deploy-the-blueprint) first. Then, you can deploy the Nvflare demo described in this document.
+
+### 1. Create custom image
+
+For this demo to work, you have to create a custom image with TensorFlow and Nvflare installed. From the main `terraform` folder:
+
+1. Build the example container image locally on your host:
+
+   ```bash
+   NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME="$(terraform output -raw container_image_repository_fully_qualified_hostname)"
+   export NVFLARE_EXAMPLE_CONTAINER_IMAGE_LOCALIZED_ID="${NVFLARE_EXAMPLE_CONTAINER_IMAGE_REPOSITORY_HOSTNAME}/$(terraform output -raw container_image_repository_name)/nvflare-tensorflow"
```

Comment:
Avoid assigning and exporting in the same instruction when you have dynamically assigned variables because the `export` command can mask potential errors that the subshell might return.

Refactor both to:

```sh
VAR="$(foo)"
export VAR
```